torch/nn/modules/module.py

```diff
@@ -419,8 +424,16 @@ def to(self, *args, **kwargs):
                 raise TypeError('nn.Module.to only accepts floating point '
                                 'dtypes, but got desired dtype={}'.format(dtype))
 
+        # memory_format handled separately as we can't call `to` operator with
+        # only memory_format, and we want to keep ability to convert layouts,
```

Comment:
I don't really understand this clause, don't we already have the ability to convert layouts, etc. without converting dtype or device?

Also, what happens if you call `to(tensor)`?  Does it convert the memory format?  My overall point is I still don't see why we don't just do the parsing correctly in _parse_to -- as this is written, I can do something like `to(tensor, memory_format=...)` but that's not the intention of the API.  But that's perfectly expressible with the python_arg_parser.

Reply:
Fixed.